openff/toolkit/topology/topology.py

```diff
@@ -1204,9 +1210,22 @@ def _invalidate_cached_properties(self):
             if "_topology_atom_index" in atom.__dict__:
                 del atom.__dict__["_topology_atom_index"]
 
-    def copy_initializer(self, other):
-        other_dict = deepcopy(other.to_dict())
-        self._initialize_from_dict(other_dict)
+    # Any particular need for this method to be public?
+    def copy_initializer(self, other: "Topology"):
+        """Initialize a new `Topology` by copying another object."""
+        self.__init__()  # type: ignore[misc]
+
+        self._aromaticity_model = other.aromaticity_model
+
+        for pair, distance in other.constrained_atom_pairs.items():
+            self.add_constraint(pair, distance)
+
+        self.box_vectors = other.box_vectors
+
+        for other_molecule in other.molecules:
+            self._add_molecule_keep_cache(other_molecule.__class__(other_molecule))  # type: ignore[call-arg]
```

Comment:
👍 I broadly agree with this approach for this PR. In the long run we'll want to think through a more robust way to enable serialization of Topologies with SimpleMolecules, but in the short run this is the best way forward.